listenbrainz/webserver/static/js/src/PinRecordingModal.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from "react";
+import { get as _get } from "lodash";
+import GlobalAppContext from "./GlobalAppContext";
+
+export type PinRecordingModalProps = {
+  recordingToPin?: Listen;
+  isCurrentUser: Boolean;
+  newAlert: (
+    alertType: AlertType,
+    title: string,
+    message: string | JSX.Element
+  ) => void;
+};
+
+export interface PinRecordingModalState {
+  blurbContent?: string;
+}
+
+export default class PinRecordingModal extends React.Component<
+  PinRecordingModalProps,
+  PinRecordingModalState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+  readonly maxBlurbContentLength = 280;
+
+  constructor(props: PinRecordingModalProps) {
+    super(props);
+    this.state = { blurbContent: undefined };
```

Comment:
this seems weird? Could we initialize it as an empty string `''` and then remove the optionality from the `PinRecordingModalState` type as well? It removes us having to work with undefined, less code is always good.

what do you think?

Reply:
Done. I agree, string `''` and undefined string are the same but this lets us remove the optionality.